diff -up rust-packaging-24/macros.d/macros.cargo.1~ rust-packaging-24/macros.d/macros.cargo
--- rust-packaging-24/macros.d/macros.cargo.1~	2023-05-09 22:40:23.529231705 +0200
+++ rust-packaging-24/macros.d/macros.cargo	2023-05-09 22:41:21.876693138 +0200
@@ -149,7 +149,8 @@ if %{__cargo_is_lib} ; then
   REG_DIR=%{buildroot}%{cargo_registry}/$CRATE_NAME-$CRATE_VERSION  \
   %{__mkdir} -p $REG_DIR                                            \
   %{__awk} -i inplace -v INPLACE_SUFFIX=.deps '/^\\\[((.+\\\.)?((dev|build)-)?dependencies|features)/{f=1;next} /^\\\[/{f=0}; !f' Cargo.toml \
-  %{__cargo} package -l | grep -w -E -v 'Cargo.(lock|toml.orig)' | xargs -d '\\\n' %{__cp} --parents -a -t $REG_DIR \
+# --allow-dirty is needed because *.cargo files don't package .git and friends \
+  %{__cargo} package --allow-dirty -l | grep -w -E -v 'Cargo.(lock|toml.orig)' | xargs -d '\\\n' %{__cp} --parents -a -t $REG_DIR \
   %{__mv} Cargo.toml{.deps,}                                        \
   %{__cp} -a Cargo.toml $REG_DIR/Cargo.toml                         \
   %{__rm} -f $REG_DIR/Cargo.toml.deps                               \
